DSCResources/MSFT_xIisFeatureDelegation/MSFT_xIisFeatureDelegation.psm1

```diff
@@ -0,0 +1,168 @@
+######################################################################################
+# DSC Resource for IIS Server level Feature Delegation
+######################################################################################
+
+data LocalizedData
+{
+    # culture="en-US"
+    ConvertFrom-StringData @'
+NoWebAdministrationModule=Please ensure that WebAdministration module is installed.
+UnableToGetConfig=Unable to get configuration data for '{0}'
+ConfigNotFound=No configuration data found for '{0}'
+ChangedMessage=Changed overrideMode for '{0}' to {1}
+'@
+}
+
+######################################################################################
+# The Get-TargetResource cmdlet.
+# This function will get the Mime type for a file extension
+######################################################################################
+function Get-TargetResource
+{
+  [OutputType([Hashtable])]
+    param
+    (        
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [String]$SectionName,
+
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateSet("Allow", "Deny")]
+        [String]$OverrideMode
+    )
+    
+    CheckIISPoshModule
+
+    [string]$oMode = GetOverrideMode -section $SectionName
+
+    if ($oMode -eq $null)
+    {
+        $ensureResult = "Absent";
+    }
+    else
+    {        
+        if ($oMode -eq $OverrideMode)
+        {
+            $ensureResult = "Present"
+        }
+        else
+        {
+            $ensureResult = "Absent";
+        }
+    }
+
+    # in case the section has not been found, $oMode will be $null
+    $getTargetResourceResult = @{SectionName = $SectionName
```

Comment:
You're still not returning $ensureResult, I would just add it to the hashtable here.